autopilot/sagemaker_autopilot_direct_marketing.ipynb

```diff
@@ -0,0 +1,647 @@
+    "plt.figure(num=None, figsize=(16, 9), dpi=160, facecolor=\"w\", edgecolor=\"k\")\n",
+    "fpr, tpr, _ = roc_curve(labels == \"yes\", test_data_preds.probability)\n",
+    "plt.plot(fpr, tpr)\n",
+    "plt.plot([0, 1], [0, 1])\n",
+    "plt.xlabel(\"False Positive Rate\")\n",
+    "plt.ylabel(\"True Positive Rate\")\n",
+    "plt.title(\"ROC Curve\")\n",
+    "\n",
+    "\n",
+    "plt.figure(num=None, figsize=(16, 9), dpi=160, facecolor=\"w\", edgecolor=\"k\")\n",
+    "precision, recall, _ = precision_recall_curve(\n",
+    "    labels == \"yes\", test_data_preds.probability\n",
+    ")\n",
+    "plt.plot(recall, precision)\n",
+    "plt.plot([0, 1], [1, 0])\n",
+    "plt.xlabel(\"Recall\")\n",
+    "plt.ylabel(\"Precision\")\n",
+    "plt.title(\"Precision-Recall Curve\")\n",
+    "plt.title(\"ROC Curve\");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Candidate Generation Notebook\n",
+    "    \n",
+    "Sagemaker AutoPilot also auto-generates a Candidate Definitions notebook. This notebook can be used to interactively step through the various steps taken by the Sagemaker Autopilot to arrive at the best candidate. This notebook can also be used to override various runtime parameters like parallelism, hardware used, algorithms explored, feature extraction scripts and more.\n",
+    "    \n",
+    "The notebook can be downloaded from the following Amazon S3 location:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "automl_job.describe_auto_ml_job()[\"AutoMLJobArtifacts\"][\n",
+    "    \"CandidateDefinitionNotebookLocation\"\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data Exploration Notebook\n",
+    "Sagemaker Autopilot also auto-generates a Data Exploration notebook, which can be downloaded from the following Amazon S3 location:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "automl_job.describe_auto_ml_job()[\"AutoMLJobArtifacts\"][\n",
+    "    \"DataExplorationNotebookLocation\"\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Cleanup\n",
+    "\n",
+    "The Autopilot job creates many underlying artifacts such as dataset splits, preprocessing scripts, or preprocessed data, etc. This code, when un-commented, deletes them. This operation deletes all the generated models and the auto-generated notebooks as well. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import boto3\n",
+    "\n",
+    "# s3 = boto3.resource('s3')\n",
+    "# bucket = s3.Bucket(bucket)\n",
+    "\n",
+    "# job_outputs_prefix = '{}/output/{}'.format(prefix,auto_ml_job_name)\n",
+    "# bucket.objects.filter(Prefix=job_outputs_prefix).delete()"
```

Comment:
Seems like including these S3 deletion steps is an implicit suggestion that all the notebook's outputs in S3 are cleaned up (which isn't true)?

Reply:
Why not true? I checked, all artifacts and models are in the path.

Reply:
It doesn't catch the input datasets though, right (I think)? Those are uploaded to `{prefix}/train` and `{prefix}/test`?. Kept as-is for now but might be nice to resolve sometime in future